### PR TITLE
chore: updated convenience methods

### PR DESCRIPTION
**Pull request by Anush008**
This PR adds convenience methods for
- Creating `PointsSelector` from a list of values.
- Creating a list of `Value` for payloads.
- Payload field selectors(Include from string[], ! operator to Exclude)
- Vector selectors

### CODE REVIEW

File: src/Qdrant.Client/Grpc/WithPayloadSelector.cs [hunk 1]
```diff
@@ -11,4 +11,39 @@ public partial class WithPayloadSelector
...
+			};
+		}
+		else
+		{
+			return new WithPayloadSelector
+			{
+				Include = new PayloadIncludeSelector { Fields = { selector.Exclude.Fields } }
+			};
+		}
+	}
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/WithPayloadSelector.cs:48]: Is there a current need for this, and is it a common operation consumers would be expected to perform? If the current need is unknown, I'd lean towards not supporting this

File: src/Qdrant.Client/Grpc/PointsSelector.cs [hunk 2]
```diff
@@ -8,12 +8,22 @@ public partial class PointsSelector
...
+	/// Implicitly converts an array of <see cref="ulong" /> ID to a new instance of <see cref="PointsSelector"/>
+	/// </summary>
+	public static implicit operator PointsSelector(ulong[] ids) =>
+		new() { Points = new PointsIdsList { Ids = { ids.Select(id => (PointId)id).ToArray() } } };
+
+	/// <summary>
+	/// Implicitly converts an array of <see cref="Guid" /> ID to a new instance of <see cref="PointsSelector"/>
+	/// </summary>
+	public static implicit operator PointsSelector(Guid[] ids) =>
+		new() { Points = new PointsIdsList { Ids = { ids.Select(id => (PointId)id).ToArray() } } };
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/PointsSelector.cs:28]: `.ToArray()` is unneccessary as collection initializer for `RepeatedField<T>` will enumerate the `Enumerable<PointId>` projected by LINQ. Removing also avoids an allocation.

```suggestion
		new() { Points = new PointsIdsList { Ids = { ids.Select(id => (PointId)id) } } };
```

File: src/Qdrant.Client/Grpc/PointsSelector.cs [hunk 3]
```diff
@@ -8,12 +8,22 @@ public partial class PointsSelector
...
 	/// </summary>
-	public static implicit operator PointsSelector(ulong id)
-		=> new() { Points = new PointsIdsList { Ids = { new PointId { Num = id } } } };
+	public static implicit operator PointsSelector(ulong id) => new ulong[] { id };
+
+	/// <summary>
+	/// Implicitly converts an array of <see cref="ulong" /> ID to a new instance of <see cref="PointsSelector"/>
+	/// </summary>
+	public static implicit operator PointsSelector(ulong[] ids) =>
+		new() { Points = new PointsIdsList { Ids = { ids.Select(id => (PointId)id).ToArray() } } };
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/PointsSelector.cs:22]: `.ToArray()` is unneccessary as collection initializer for `RepeatedField<T>` will enumerate the `Enumerable<PointId>` projected by LINQ. Removing also avoids an allocation.

```suggestion
		new() { Points = new PointsIdsList { Ids = { ids.Select(id => (PointId)id) } } };
```

File: src/Qdrant.Client/Grpc/PointsSelector.cs [hunk 4]
```diff
@@ -8,12 +8,22 @@ public partial class PointsSelector
 	/// <summary>
 	/// Implicitly converts a <see cref="Guid" /> ID to a new instance of <see cref="PointsSelector"/>
 	/// </summary>
-	public static implicit operator PointsSelector(Guid id)
-		=> new() { Points = new PointsIdsList { Ids = { new PointId { Uuid = id.ToString() } } } };
+	public static implicit operator PointsSelector(Guid id) => new Guid[] { id };
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/PointsSelector.cs:11]: Suggest keeping this as- it's a little more obvious than chaining on to another implicit overload, and avoids the unnecessary array allocation.

File: src/Qdrant.Client/Grpc/PointsSelector.cs [hunk 5]
```diff
@@ -8,12 +8,22 @@ public partial class PointsSelector
...
 
 	/// <summary>
 	/// Implicitly converts a <see cref="ulong" /> ID to a new instance of <see cref="PointsSelector"/>
 	/// </summary>
-	public static implicit operator PointsSelector(ulong id)
-		=> new() { Points = new PointsIdsList { Ids = { new PointId { Num = id } } } };
+	public static implicit operator PointsSelector(ulong id) => new ulong[] { id };
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/PointsSelector.cs:16]: Suggest keeping this as- it's a little more obvious than chaining on to another implicit overload, and avoids the unnecessary array allocation.

File: src/Qdrant.Client/Grpc/WithVectorsSelector.cs [hunk 6]
```diff
@@ -11,4 +11,12 @@ public partial class WithVectorsSelector
 	/// <param name="enable">If <code>true</code> return all vectors, if <code>false</code> then none</param>
 	/// <returns>a new instance of <see cref="WithPayloadSelector"/></returns>
 	public static implicit operator WithVectorsSelector(bool enable) => new() { Enable = enable };
+
+	/// <summary>
+	/// Implicitly converts an array <see cref="string"/> to a new instance of <see cref="WithPayloadSelector"/>
+	/// </summary>
+	/// <param name="vectors">List of vector names to return.</param>
```
Comment (russcam) [inline src/Qdrant.Client/Grpc/WithVectorsSelector.cs:18]:
```suggestion
	/// <param name="vectors">List of vector names to include in the response.</param>
```

**Review by russcam — CHANGES_REQUESTED** [review]

File: src/Qdrant.Client/Grpc/WithPayloadSelector.cs (hunk 1 above)
Comment (Anush008) [reply src/Qdrant.Client/Grpc/WithPayloadSelector.cs]: https://github.com/qdrant/qdrant-dotnet/pull/41/commits/07bea9863f22f407534d184fdc6f8991e4ffd2cf

File: src/Qdrant.Client/Grpc/PointsSelector.cs (hunk 4 above)
Comment (Anush008) [reply src/Qdrant.Client/Grpc/PointsSelector.cs]: https://github.com/qdrant/qdrant-dotnet/pull/41/commits/07bea9863f22f407534d184fdc6f8991e4ffd2cf

File: src/Qdrant.Client/Grpc/PointsSelector.cs (hunk 5 above)
Comment (Anush008) [reply src/Qdrant.Client/Grpc/PointsSelector.cs]: https://github.com/qdrant/qdrant-dotnet/pull/41/commits/07bea9863f22f407534d184fdc6f8991e4ffd2cf

**Review by russcam — APPROVED** [review]

**Pull request merged**